src/NodaTime.Test/DateIntervalTest.cs

```diff
@@ -331,6 +332,30 @@ public void Union(string first, string second, string expected)
             });
         }
 
+        [Test]
```

Comment:
Add a test to verify we can iterate over a minimal (one-day) `LocalDateInterval` too?

Reply:
Good call. Rewritten to be a single data-driven test :)